sph/include/sph/particles_data_gpu.cuh

```diff
@@ -184,6 +184,23 @@ public:
         return 0;
     }
 
+    util::array<std::size_t, 5> memStats()
+    {
+        auto        data_ = data();
+        std::size_t sumOfSize{0}, sumOfCap{0};
+        for (size_t i = 0; i < data_.size(); ++i)
+        {
+            sumOfSize +=
+                std::visit([]<class V>(V* arg) { return sizeof(typename V::value_type) * arg->size(); }, data_[i]);
+            sumOfCap +=
+                std::visit([]<class V>(V* arg) { return sizeof(typename V::value_type) * arg->capacity(); }, data_[i]);
+        }
+
+        std::size_t free{0}, total{0};
+        cudaMemGetInfo(&free, &total);
+        return {size(), sumOfSize, sumOfCap, free, total};
+    }
+
```

Comment:
Does it make sense/Is it possible to do a bit of refactoring of this function with the corresponding one of particles_data.hpp? If I'm not wrong their implementation is similar except for the call to cudaMemGetInfo.